compiler/rustc_session/src/options.rs

```diff
@@ -20,42 +20,80 @@ use std::num::NonZeroUsize;
 use std::path::PathBuf;
 use std::str;
 
-macro_rules! hash_option {
-    ($opt_name:ident, $opt_expr:expr, $sub_hashes:expr, [UNTRACKED]) => {{}};
-    ($opt_name:ident, $opt_expr:expr, $sub_hashes:expr, [TRACKED]) => {{
+macro_rules! insert {
+    ($opt_name:ident, $opt_expr:expr, $sub_hashes:expr) => {
         if $sub_hashes
             .insert(stringify!($opt_name), $opt_expr as &dyn dep_tracking::DepTrackingHash)
             .is_some()
         {
             panic!("duplicate key in CLI DepTrackingHash: {}", stringify!($opt_name))
         }
-    }};
+    };
+}
+
+macro_rules! hash_all_tracked {
+    ($opt_name:ident, $opt_expr:expr, $sub_hashes:expr, [UNTRACKED]) => {{}};
+    ($opt_name:ident, $opt_expr:expr, $sub_hashes:expr, [TRACKED]) => {{ insert!($opt_name, $opt_expr, $sub_hashes) }};
+    ($opt_name:ident, $opt_expr:expr, $sub_hashes:expr, [TRACKED_NO_CRATE_HASH]) => {{ insert!($opt_name, $opt_expr, $sub_hashes) }};
+    ($opt_name:ident, $opt_expr:expr, $sub_hashes:expr, [SUBSTRUCT]) => {{}};
+}
+
+macro_rules! hash_crate_hash {
```

Comment:
Is it possible to unify the `hash_all_tracked!` and the `hash_crate_hash!` macros by adding `for_crate_hash` as a parameter and then use `if` in the `[TRACKED_NO_CRATE_HASH]` case? I'll leave it up to you whether you want to look into this.

Reply:
Sure thing, done.